### PR TITLE
[Doc] Document the top-k default scheduling strategy

### LINKED ISSUE #32154: [Ray 2.3] update doc about top-k scheduling policy
### Description

Update doc to reflect the change of default scheduling strategy in https://github.com/ray-project/ray/pull/31868

### Link

_No response_

### PR DESCRIPTION
**Pull request by jjyao**
Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>

## Why are these changes needed?
Document the top-k default scheduling strategy

## Related issue number
Closes #32154

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-core/scheduling/index.rst [hunk 1]
```diff
@@ -37,8 +37,11 @@ Currently the supported strategies are the followings.
 "DEFAULT"
 ~~~~~~~~~
 
-``"DEFAULT"`` is the default strategy used by Ray. With the current implementation, Ray will try to pack tasks or actors on nodes
-until the resource utilization is beyond a certain threshold and spread them afterwards.
+``"DEFAULT"`` is the default strategy used by Ray.
+With the current implementation, Ray calculates a score for each node based on the node's logical resource utilization.
+If the utilization is below a threshold, the score is 0. Otherwise, the score is the resource utilization itself (score 1 means the node is fully utilized).
+Ray randomly picks a node from the top k nodes with the lowest scores as the best node for scheduling.
```
[the remark this answers is not in the document]
Comment (scv119) [reply doc/source/ray-core/scheduling/index.rst:43]: do we want to also document the two options control the top_k?

File: doc/source/ray-core/scheduling/index.rst [hunk 2]
```diff
@@ -37,8 +37,13 @@ Currently the supported strategies are the followings.
 "DEFAULT"
 ~~~~~~~~~
 
-``"DEFAULT"`` is the default strategy used by Ray. With the current implementation, Ray will try to pack tasks or actors on nodes
-until the resource utilization is beyond a certain threshold and spread them afterwards.
```
Comment (stephanie-wang) [inline doc/source/ray-core/scheduling/index.rst:41]: Can we keep some high-level description of the policy? The new version of the text is very low-level and it's a bit hard to understand the high-level behavior, which is what most users will care about.

File: doc/source/ray-core/scheduling/index.rst [hunk 3]
```diff
@@ -37,8 +37,18 @@ Currently the supported strategies are the followings.
...
+``"DEFAULT"`` is the default strategy used by Ray.
+Ray schedules tasks or actors onto a group of the top k nodes.
+Specially, the nodes are sorted to first favor those that already have tasks or actors scheduled (for locality),
```
Comment (stephanie-wang) [inline doc/source/ray-core/scheduling/index.rst:42]:
```suggestion
Specifically, the nodes are sorted to first favor those that already have tasks or actors scheduled (for locality),
```

File: doc/source/ray-core/scheduling/index.rst [hunk 4]
```diff
@@ -37,8 +37,18 @@ Currently the supported strategies are the followings.
...
+then to favor those that have low resource utilization (for load balancing).
+Within the top k group, nodes are chosen randomly to further improve load-balancing and mitigate delays from cold-start in large clusters.
+
+Implementation-wise, Ray calculates a score for each node in a cluster based on the utilization of its logical resources.
+If the utilization is below a threshold (controlled by env var ``RAY_scheduler_spread_threshold``, default is 0.5), the score is 0,
```
Comment (stephanie-wang) [inline doc/source/ray-core/scheduling/index.rst:47]:
```suggestion
If the utilization is below a threshold (controlled by the OS environment variable ``RAY_scheduler_spread_threshold``, default is 0.5), the score is 0,
```

File: doc/source/ray-core/scheduling/index.rst [hunk 5]
```diff
@@ -37,8 +37,18 @@ Currently the supported strategies are the followings.
...
+otherwise it is the resource utilization itself (score 1 means the node is fully utilized).
+Ray selects the best node for scheduling by randomly picking from the top k nodes with the lowest scores.
+The value of ``k`` equals to the max of (# nodes in the cluster * ``RAY_scheduler_top_k_fraction`` env var) and ``RAY_scheduler_top_k_absolute`` env var.
```
Comment (stephanie-wang) [inline doc/source/ray-core/scheduling/index.rst:50]:
```suggestion
The value of ``k`` is the max of (number of nodes in the cluster * ``RAY_scheduler_top_k_fraction`` environment variable) and ``RAY_scheduler_top_k_absolute`` environment variable.
```

**Review by stephanie-wang — APPROVED** [review]

**Pull request merged**